module/Category/src/Controller/CategoryController.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Category\Controller;
+
+use Category\Model\Category;
+use Doctrine\ORM\EntityManager;
+use Product\model\Product as ProductModel;
+use Zend\Mvc\Controller\AbstractActionController;
+use Zend\View\Model\ViewModel;
+use Category\Model\Category as CategoryModel;
+class CategoryController extends AbstractActionController
+{
+    /**
+     * Entity manager.
+     * @var Doctrine\ORM\EntityManager
+     */
+    private $entityManager;
+
+    public function __construct($serviceManager)
```

Comment:
Rather than injecting the $serviceManager, you should inject directly the services you need - in this case $entityManager, although really you should inject own service (`CategoryModel`) in this module and inject that (with $entityManager injected into that service, instead of into this controller).

Injecting the $serviceManager is considered to be an anti-pattern.